pkg/controller/controller_instance.go

```diff
@@ -70,6 +70,127 @@ func (c *controller) instanceUpdate(oldObj, newObj interface{}) {
+		PlanID:            servicePlan.ExternalID,
+		AcceptsIncomplete: true,
+	}
+
+	glog.V(4).Infof("Deprovisioning Instance %v/%v of ServiceClass %v at Broker %v", instance.Namespace, instance.Name, serviceClass.Name, brokerName)
+	response, respCode, err := brokerClient.DeleteServiceInstance(instance.Spec.ExternalID, request)
+
+	if err != nil {
+		s := fmt.Sprintf(
+			"Error deprovisioning Instance \"%s/%s\" of ServiceClass %q at Broker %q with status code %d: %s",
+			instance.Namespace,
+			instance.Name,
+			serviceClass.Name,
+			brokerName,
+			respCode,
+			err,
+		)
+		glog.Warning(s)
+		c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionUnknown,
+			errorDeprovisionCalledReason,
+			"Deprovision call failed. "+s)
+		c.recorder.Event(instance, api.EventTypeWarning, errorDeprovisionCalledReason, s)
+		return err
+	}
+
+	if respCode == http.StatusAccepted {
+		glog.V(5).Infof("Received asynchronous de-provisioning response for Instance %v/%v of ServiceClass %v at Broker %v: response: %v", instance.Namespace, instance.Name, serviceClass.Name, brokerName, response)
+		if response.Operation != "" {
+			instance.Status.LastOperation = &response.Operation
+		}
+
+		// Tag this instance as having an ongoing async operation so we can enforce
+		// no other operations against it can start.
+		instance.Status.AsyncOpInProgress = true
+
+		err := c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			asyncDeprovisioningReason,
+			asyncDeprovisioningMessage,
+		)
+		if err != nil {
+			return err
+		}
+	} else if respCode == http.StatusOK {
+		err := c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			successDeprovisionReason,
+			successDeprovisionMessage,
+		)
+		if err != nil {
+			return err
+		}
+		// Clear the finalizer
+		finalizers.Delete(finalizerToken)
+		if err = c.updateInstanceFinalizers(instance, finalizers.List()); err != nil {
+			return err
+		}
+		c.recorder.Event(instance, api.EventTypeNormal, successDeprovisionReason, successDeprovisionMessage)
+		glog.V(5).Infof("Successfully deprovisioned Instance %v/%v of ServiceClass %v at Broker %v", instance.Namespace, instance.Name, serviceClass.Name, brokerName)
+	} else {
+		// the broker returned a failure response
+		errorDeprovisionCalledMessage := fmt.Sprintf("deprovision call failed")
+		err := c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			errorDeprovisionCalledReason,
+			errorDeprovisionCalledMessage,
+		)
+		if err != nil {
+			return err
+		}
+		c.recorder.Eventf(instance, api.EventTypeWarning, errorDeprovisionCalledReason, errorDeprovisionCalledMessage)
+	}
+	return nil
+}
+
 // reconcileInstance is the control-loop for reconciling Instances.
 func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 
```

Comment:
Signpost, not an issue with your PR: we will need to recalculate the checksums here eventually based on the total parameter payload that we would send to broker, which may include stuff that is not in the spec (ie, content of a secret that may have changed).